### PR TITLE
Refactor SMTP and add rudimentary email+attachment construction

### PR DESCRIPTION
**Pull request by eyj**
This completes a part of #1

### CODE REVIEW

File: email-client/mail.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	checkedWrite(mail, "X-Mailer: github.com/TNG/gpg-validation-server\r\n")
+
+	// Now follow the MIME Headers
+	checkedWrite(mail, "Mime-Version: 1.0 (Golang 1.6)\r\n")
+	// checkedWrite(mail, "Content-Type: multipart/encrypted; boundary=\"" + mw.Boundary() + "\"; protocol=\"application/pgp-encrypted\";\r\n")
+	checkedWrite(mail, "Content-Type: multipart/plain; boundary=\""+mw.Boundary()+"\";\r\n")
+	checkedWrite(mail, "\r\n")
+	checkedWrite(mail, "This is an OpenPGP/MIME encrypted message (RFC 2440 and 3156)\r\n")
+
+	pw, _ := mw.CreatePart(part1.Headers)
```
Comment (Pharb) [inline email-client/mail.go]: I would love a little bit more readable variable names 😄 
`partWriter, _ := multipartWriter.CreatePart(...)`

File: email-client/mail.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+func checkedWrite(buffer bytes.Buffer, s string) {
+	_, err := buffer.WriteString(s)
+	if err != nil {
+		panic(err)
+	}
+}
+
+// ConstructEmail constructs an email to the given recipient consisting of the two MIMEParts
+func ConstructEmail(recipient string, part1, part2 MIMEPart) string {
```
Comment (Pharb) [inline email-client/mail.go]: Do we only need two `MIMEPart`s? Wouldn't using var args be trivial here?

File: email-client/mail.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	checkedWrite(mail, "Mime-Version: 1.0 (Golang 1.6)\r\n")
+	// checkedWrite(mail, "Content-Type: multipart/encrypted; boundary=\"" + mw.Boundary() + "\"; protocol=\"application/pgp-encrypted\";\r\n")
+	checkedWrite(mail, "Content-Type: multipart/plain; boundary=\""+mw.Boundary()+"\";\r\n")
+	checkedWrite(mail, "\r\n")
+	checkedWrite(mail, "This is an OpenPGP/MIME encrypted message (RFC 2440 and 3156)\r\n")
+
+	pw, _ := mw.CreatePart(part1.Headers)
+	_, err := pw.Write(part1.Message)
+	if err != nil {
+		panic(err)
```
Comment (Pharb) [inline email-client/mail.go]: Why `panic` and not just return the `err`?

File: email-client/smtp_test.go [hunk 4]
```diff
@@ -10,9 +10,31 @@ import (
...
+		log.Fatal(envelope.Message)
+	}
+	return nil
+}
+
+func TestEmail(t *testing.T) {
+	mailer := mockSendMailer{"sender@localhost.local", "recipient@localhost.local", "Hey, you!"}
+	mail := MailEnvelope{"sender@localhost.local", "recipient@localhost.local", "Hey, you!"}
+	_ = mailer.SendMail(mail)
+}
```
Comment (MartinAltmayer) [inline email-client/smtp_test.go]: What exactly does this function test?

File: email-client/smtp_test.go [hunk 5]
```diff
@@ -10,9 +10,31 @@ import (
...
+	if envelope.Sender != mailer.expectedSender {
+		log.Fatal(envelope.Sender)
+	}
+	if envelope.Recipient != mailer.expectedRecipient {
+		log.Fatal(envelope.Recipient)
+	}
+	if envelope.Message != mailer.expectedMessage {
+		log.Fatal(envelope.Message)
+	}
+	return nil
```
Comment (Pharb) [inline email-client/smtp_test.go]: Why return `nil`?

**Comment by Pharb** [comment]
LGTM 👍

File: email-client/smtp_test.go [hunk 6]
```diff
@@ -29,9 +51,16 @@ func TestSendEmail(t *testing.T) {
 			log.Fatal(err)
 		}
 	}()
+}
+
+func TestSMTPSendMailer(t *testing.T) {
+	resultChannel := make(chan string)
```
Comment (MartinAltmayer) [inline email-client/smtp_test.go]: Is anyone reading from this channel?

File: email-client/smtp_test.go (hunk 5 above)
Comment (eyj) [inline email-client/smtp_test.go]: No error occurred?

File: email-client/smtp_test.go (hunk 4 above)
Comment (eyj) [inline email-client/smtp_test.go]: Not much. I just wanted to write a Mock send mailer here ;)

File: email-client/mail.go (hunk 3 above)
Comment (eyj) [inline email-client/mail.go]: The function does not return an error.

File: email-client/mail.go (hunk 2 above)
Comment (eyj) [inline email-client/mail.go]: Done.

File: email-client/mail.go (hunk 1 above)
Comment (eyj) [inline email-client/mail.go]: Done.

**Pull request closed without merging**